stan/math/opencl/matrix_cl.hpp

```diff
@@ -259,12 +260,12 @@ class matrix_cl<T, enable_if_arithmetic<T>> {
        * on the device until we are sure that the data
        * is finished transfering
        */
-      cl::Event write_event;
-      queue.enqueueWriteBuffer(
-          buffer_cl_, CL_FALSE, sizeof(double) * offset_size,
-          sizeof(double) * rows_, A[i].data(), nullptr, &write_event);
-      this->add_write_event(write_event);
+      queue.enqueueWriteBuffer(buffer_cl_, CL_FALSE,
+                               sizeof(double) * offset_size,
+                               sizeof(double) * rows_, A[i].data(),
+                               &this->read_events(), &write_event);
```

Comment:
This is a constructor - this matrix was not used before. So it has no read events and this change is unnecessary.